crates/rune/src/languageserver/state.rs

```diff
@@ -250,6 +250,34 @@ impl<'a> State<'a> {
         Some(results)
     }
 
+    pub(super) fn format(&mut self, uri: &Url) -> Option<TextEdit> {
+        let sources = &mut self.workspace.sources;
+        tracing::trace!(uri = ?uri, uri_exists = sources.get(uri).is_some());
+        let workspace_source = sources.get_mut(uri)?;
+
+        // Not sure how to get source of `workspace_source`
+        let source = crate::source::Source::new(uri, workspace_source.content.to_string());
+        let formatted = crate::fmt::layout_source(&source).ok()?;
+        let formatted = String::from_utf8(formatted).expect("format only produces valid utf8");
```

Comment:
Err instead of panic. The former is handled by the server.

Reply:
Right, i just expected fmt to check this already meaning that this error can never happen.

Reply:
Only thing I see would be to return `String` instead of `Vec<u8>`?

Reply:
Yeah I get it. That might be true today, but it's a big invariant to remember and maintain. The outcome of a panic is the same as erroring (language server goes down), but with worse diagnostics.